spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/EnvironmentEntryInitializingTreeMap.java

```diff
@@ -52,18 +52,18 @@
 	/**
 	 * Constructs the map.
 	 * 
-	 * @param applicationContext the application context that supplies the property values
+	 * @param environment the environment that supplies the default property values
 	 * @param entryClass the entry class
 	 * @param defaultsPrefix the prefix for initializing the properties
 	 * @param delegate the actual map that stores the values
 	 */
-	public EnvironmentEntryInitializingTreeMap(ConfigurableApplicationContext applicationContext, Class<T> entryClass,
+	public EnvironmentEntryInitializingTreeMap(ConfigurableEnvironment environment, Class<T> entryClass,
 			String defaultsPrefix, Map<String, T> delegate) {
```

Comment:
Well, if we made that a prefix then the class implementation would be effectively coupled to it. Suppose we'd want to reuse it for Rabbit, Kafka or other kinds of defaults?